### PR TITLE
Provide our own ssl certs on Windows Ruby

### LINKED ISSUE #2402: `rake bootstrap` on Windows Ruby cannot 'gem install minitar'
Bootstrapping goes: download jruby, unpack jruby, relaunch under jruby.

However, with the Windows Ruby installer, it ships with a broken rubygems configuration such that `gem install` cannot work due to SSL cert trust problems.

```
ERROR:  Could not find a valid gem 'minitar' (>= 0), here is why:
          Unable to download data from https://rubygems.org/ - SSL_connect returned=1 errno=0 state=SSLv3 read server ce
rtificate B: certificate verify failed (https://api.rubygems.org/latest_specs.4.8.gz)
Installation of minitar failed
```

The recommended workaround for each user is this: https://gist.github.com/luislavena/f064211759ee0f806c88

However, I think this is a bad workaround. It requires every dev on windows to do this "hunt for the ruby directory, download this random cert file, and hope you did it right". I want to automate the workaround, but I am at present unable to find a solution: https://gist.github.com/jordansissel/76b47be00509c5b6f5a0

### PR DESCRIPTION
**Pull request by jordansissel**
This fixes a bug where Windows Ruby ships with insufficient ssl
certificate information and this causes gem installations to fail. This
patch is only applied under Windows ruby because JRuby is not affected
by it.

Fixes #2402

### CODE REVIEW

**Comment by jordansissel** [comment]
To test, running `rake bootstrap` from a clean checkout (or purge `vendor` and `build`). I tested on ruby 2.1.5p273 (2014-11-13 revision 48405) [x64-mingw32] (Windows Ruby)

**Comment by jordansissel** [comment]
Full test: 
```
rake bootstrap
rake test:prep
bin/logstash rspec

---
Finished in 2.3 seconds
252 examples, 0 failures
```

File: tools/ca/Class3PublicPrimaryCertificationAuthority.pem [hunk 1]
```diff
@@ -0,0 +1,14 @@
+-----BEGIN CERTIFICATE-----
```
Comment (jordansissel) [inline tools/ca/Class3PublicPrimaryCertificationAuthority.pem]: This file came with Windows Ruby.

File: tools/ca/Entrust_net-Secure-Server-Certification-Authority.pem [hunk 2]
```diff
@@ -0,0 +1,90 @@
+This CA certificate is for verifying HTTPS connection to;
```
Comment (jordansissel) [inline tools/ca/Entrust_net-Secure-Server-Certification-Authority.pem]: This file came with Windows Ruby

File: tools/ca/EntrustnetSecureServerCertificationAuthority.pem [hunk 3]
```diff
@@ -0,0 +1,28 @@
+-----BEGIN CERTIFICATE-----
```
Comment (jordansissel) [inline tools/ca/EntrustnetSecureServerCertificationAuthority.pem]: This file came with Windows Ruby

File: tools/ca/GeoTrustGlobalCA.pem [hunk 4]
```diff
@@ -0,0 +1,20 @@
+-----BEGIN CERTIFICATE-----
```
Comment (jordansissel) [inline tools/ca/GeoTrustGlobalCA.pem]: Also came with Windows Ruby

**Comment by jordansissel** [comment]
cacert.pem came from curl's ca cert listing.

File: tools/ca/GeoTrustGlobalCA.pem (hunk 4 above)
Comment (ph) [inline tools/ca/GeoTrustGlobalCA.pem]: :8ball: says you are working on windows support..

Comment (jordansissel) [inline tools/ca/GeoTrustGlobalCA.pem]: who gave you this :8ball: ?!@?

Comment (ph) [inline tools/ca/GeoTrustGlobalCA.pem]: It came with my Windows ME CD.

Comment (untergeek) [inline tools/ca/GeoTrustGlobalCA.pem]: I need an image of Mönch's "Scream" here...

Comment (untergeek) [inline tools/ca/GeoTrustGlobalCA.pem]: The Scream

**Comment by suyograo** [comment]
@jsvd review please?

**Comment by suyograo** [comment]
tested on windows. this definitely fixed the `rake bootstrap`

**Pull request closed without merging**